Sources/Leaf/Tag/TagError.swift

```diff
@@ -0,0 +1,7 @@
+enum TagError: Error {
```

Comment:
Any thoughts on making this public? Would save anyone writing their own custom Tag having to create their own `Error` and would make it easier for anyone catching stuff?